### DISCUSSION TITLE
How to wait some time in raylib?

### DISCUSSION THREAD

**Discussion by wlxmhls** [raysan5/raylib#2537] [Q&A] [answered]
Hello,

I am a beginner. Is there time wait function in raylib, like SDL's SDL_Delay function?
Thanks

**Comment by quantumedbox** [comment, marked as answer]
`rcore.c` defines:
```c
void WaitTime(double seconds)
```

**Reply by wlxmhls** [reply]
Thank you. 
But this function is not listed in the official cheatsheet. Maybe it is NOT exposed to users?
Or any documentation list the raylib API functions, like 'void WaitTime(double seconds)'?

**Reply by wlxmhls** [reply]
emmm. I found WaitTime function in raylib.h...

**Reply by Peter0x44** [reply]
This is almost certainly *NOT* what you want to do.
This is equivalent to calling sleep, it will block your whole main loop, and the OS will probably think your program crashed.

raylib is not quite like a game engine, which runs your code. Your code is running raylib.

What you really want here, is a timer.
See [how do I make a timer?](https://github.com/raysan5/raylib/wiki/Frequently-Asked-Questions#how-do-i-make-a-timer) on the raylib FAQ.

What you should do is start a timer, keep running your main loop, and then react and do something when the timer is elapsed.

**Reply by quantumedbox** [reply]
OP asked about alternative to `SDL_Delay`, which is a blocking function.

**Comment by royqh1979** [comment]
you can use the counter of the mainloop as a timer.

If you SetTargetFPS(60), then each running of mainloop means 1/60 second passed.

**Comment by wlxmhls** [comment]
Thanks!
